course_flow/context_processors.py

```diff
@@ -5,13 +5,17 @@
 
 
 def update_notifications(request):
-    updates = UpdateNotification.objects.order_by("-created_on")
-    if updates.count() > 0:
-        last_update = updates.first()
-        last_update_serialized = (
-            JSONRenderer()
-            .render(UpdateNotificationSerializer(last_update).data)
-            .decode("utf-8")
-        )
-        return {"update_notifications": last_update_serialized}
+    try:
+        if "course-flow" in request.path:
+            updates = UpdateNotification.objects.order_by("-created_on")
+            if updates.count() > 0:
+                last_update = updates.first()
```

Comment:
Likely silly questions: Do we always only fetch first or is there only ever one (i.e. notifications are deleted once seen)?  If there are multiple, how does user get _second_ notification?

Reply:
We are fetching the most recent. Note the order_by("-created_on").

Reply:
I'm guessing you didn't notice because I did this in a clunky way, if so feel free to suggest something cleaner!

Reply:
My question is: could there ever be a situation where a user needs more than just the most recent notification?  Also, since you always want most recent, would `.last()` give the same object (and avoid the expensive ordering op)?

Reply:
Hm, yeah I guess a straight .last() would also give the same object since it'll by default be ordered by pk right? And no, there's no situation where a user should get more than the most recent one (at this time).